app/objects/c_ability.py

```diff
@@ -144,25 +100,57 @@ async def which_plugin(self):
                 return plugin
         return None
 
-    def replace_cleanup(self, encoded_cmd, payload):
-        decoded_cmd = b64decode(encoded_cmd).decode('utf-8', errors='ignore').replace('\n', '')
-        decoded_cmd = decoded_cmd.replace(self.RESERVED['payload'], payload)
-        return decoded_cmd
+    def find_executor(self, name, platform):
+        return self._executor_map.get(self._make_executor_map_key(name, platform))
+
+    def find_executors(self, names, platform):
+        """Find executors for matching platform/executor names
+
+        Only the first instance of a matching executor will be returned,
+            as there should not be multiple executors matching a single
+            platform/executor name pair.
+
+        :param names: Executors to search. ex: ['psh', 'cmd']
+        :type names: list(str)
+        :param platform: Platform to search. ex: windows
+        :type platform: str
+        :return: List of executors ordered based on ordering of `names`
+        :rtype: list(Executor)
+        """
+        executors = []
+        seen_names = set()
+        for name in names:
+            if name in seen_names:
+                continue
+            seen_names.add(name)
+
+            if self._make_executor_map_key(name, platform) in self._executor_map:
+                executors.append(self.find_executor(name, platform))
```

Comment:
Could this if statement be written as:

```python
executor = self.find_executor(name, platform)

if executor:
    executors.append(executor)

```

At a glance it looks the condition is repeating the work of `find_executor`.